backend/src/main/java/storage/repository/GenericRepository.java

```diff
@@ -0,0 +1,240 @@
+package storage.repository;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import model.DataModel;
+import storage.DBInterface;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Basic repository class that can take any model.
+ *
+ * Several things are presumed:
+ * - The model must have a default constructor
+ * - The model must set default values
+ * - The model must only use int, boolean or string (currently)
+ */
+public class GenericRepository <T extends DataModel> {
+
+    private final Gson gson;
+    private final Class<T> type;
+    private DBInterface dbInterface;
+
+    public GenericRepository(Class<T> type, DBInterface dbInterface){
+        this.gson = new Gson();
+        this.dbInterface = dbInterface;
+        this.type = type;
+    }
+
+    /***
+     * gets all objects matching given type
+     * @return
+     */
+    public List<T> getObjects() {
+        String query = "SELECT * FROM "+ type.getSimpleName();
+        System.out.println("Running query: "+query);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Running query: " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)